### ISSUE TITLE
Comments on "Make a 3D HoloLens app launcher appear above the launcher controls"

### ISSUE THREAD

**Issue by LocalJoost** [LocalJoost/BlogComments#412]
Original article: https://localjoost.github.io/Make-a-3D-HoloLens-app-launcher-appear-above-the-launcher-controls/

**Comment by ArnaudBBRI** [comment]
Hello, 

Thanks for this tip!
May I ask what version of Visual Studio you tested this on ? I'm struggling to get this working on VS 2022. As such the app builds, but no app launcher.. I do see the glb model is placed into my project/Assets/AppLauncherModel.glb after the Unity build, and this glb file is indeed the one I placed in MRTK3 > Build Settings:

![image](https://github.com/user-attachments/assets/28043ad9-65c9-44b4-93be-5fa146b22b26)

![image](https://github.com/user-attachments/assets/d7a9ccc5-130a-4342-b877-90155de0157a)

But yeah, no launcher (and no icon, as well) on my Hololens2.

I also see [here](https://learn.microsoft.com/en-us/windows/mixed-reality/distribute/implementing-3d-app-launchers) that the glb asset must be marked as "Content". The picture show in that article is slightly outdated since I have the following:

![image](https://github.com/user-attachments/assets/22117b68-6130-49fe-aeda-5339b492ba99)

If I select anything else than "Does not participate in build" for the Item Type, then I end up with a build error, complaining about

> error MSB4066: The attribute "xmlns" in element <FileType> is unrecognized.

I think I can remove that xmlns reference for VS 2017 and above (as mentioned [here](https://learn.microsoft.com/en-us/visualstudio/msbuild/msbuild-project-file-schema-reference?view=vs-2022), but then I end up with the build error

> Assets\AppLauncherModel.glb : fatal error LNK1107: invalid or corrupt file: cannot read at 0x9218

And well, beyond that is out of reach with my modest capabilities.. 😄 

Any idea ?
Thanks a lot!
Arnaud

**Comment by LocalJoost** [comment]
@ArnaudBBRI what version of Unity do you use? Also - be aware that there are severe limitations as to what GLB files you can use. They can only be like 10000 triangles. I would suggest trying to use a model that we _know_ works - like the MRTK model in my article. For the rest, [have a look at this](https://learn.microsoft.com/en-us/windows/mixed-reality/distribute/creating-3d-models-for-use-in-the-windows-mixed-reality-home#triangle-counts-and-levels-of-detail-lods?WT.mc_id=WDIT-MVP-4034925)

**Comment by ArnaudBBRI** [comment]
I use Unity 2021.3.23f1
My 3D model was created with Blender, no more than ~500 triangles, and simple color materials (2 materials). I'll try with the MRTK 3D logo, but where did you find it exactly ? Is it in the MixedRealityToolkit-Unity [github](https://github.com/MixedRealityToolkit/MixedRealityToolkit-Unity) ?

Also note, I had to use the Unity glTFast package in order to import the .glb in my project and it being accepted by the MRTK3 Build Settings in Unity.

Thanks!

**Comment by LocalJoost** [comment]
@ArnaudBBRI Yes, I found it in here. https://github.com/MixedRealityToolkit/MixedRealityToolkit-Unity/blob/main/org.mixedrealitytoolkit.extendedassets/Models/MRTK_Logo.glb. If you have installed the extended assets package it's already in your app

**Comment by ArnaudBBRI** [comment]
Ok, tested it. No more luck with this MRTK logo.
I also tried to set the "Excluded from build" setting (that I didn't notice previously, so it was left empty) to "No"

![image](https://github.com/user-attachments/assets/5e0aba33-758a-4e33-a013-20a51ff031bc)

Which modifies the vcxproj file like this:

![image](https://github.com/user-attachments/assets/03e6d179-9b41-453a-b2fd-df2773bbe5ed)

...but it doesn't help either.
The only thing I was thinking of is that in [here ](https://learn.microsoft.com/en-us/windows/mixed-reality/distribute/implementing-3d-app-launchers) they mark the File Type as Document, which is an option that no longer exists (at least in my VS 2022):

![buildsetting-content-300px](https://github.com/user-attachments/assets/2d396a26-5971-4a8a-81a4-e0ccd1d7efe4)

But I digged further, and in fact I do have a compiled scene from the MRTK samples, on the same Unity version, same VS version, and for which Item Type is "Do not participate in build", and in that case it works.

I compared both appxmanifests, the only noticeable difference between the working project (MRTK sample) and the non-working project (mine) was the supported rotations (only landscape for the MRTK sample, multiple for mine). I changed that, didn't help.
Also compared the vcxproject files, nothing to be seen there either.
